php-frontend/src/main/java/org/sonar/DurationStatistics.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar;
+
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.text.NumberFormat;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Supplier;
+import org.sonar.api.config.Configuration;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+
+public class DurationStatistics {
+
+  private static final Logger LOG = Loggers.get(DurationStatistics.class);
+
+  private static final String PROPERTY_KEY = "sonar.php.duration.statistics";
+
+  private final Map<String, AtomicLong> stats = new ConcurrentHashMap<>();
+
+  private final boolean recordStat;
+
+  public DurationStatistics(Configuration config) {
+    recordStat = config.getBoolean(PROPERTY_KEY).orElse(false);
+  }
+
+  public void time(String id, Runnable runnable) {
```

Comment:
If I'm not mistaken, this method is never used (but tested). I think we can remove it if that is the case.